File: Data Structures & Algorithms/rotate-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head)return nullptr;
        int n=0;
        ListNode* temp=head;
        ListNode* tail=new ListNode();
        while(temp){
            tail=temp;
            n+=1;
            temp=temp->next;
        }
        
        k=k%n;
        if(k==0)return head;
        ListNode* newT=head;
        for(int i=0;i<n-k-1;i++){
            newT=newT->next;
        }
        ListNode*newH=newT->next;
        newT->next=nullptr;
        tail->next=head;

        return newH;

    }
};

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        slow, fast = head, head

        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next

        prev = None
        curr = slow
        while curr:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp

        fHead = head
        sHead = prev

        while sHead:
            if sHead.val != fHead.val:
                return False
            fHead = fHead.next
            sHead = sHead.next

        return True
